python/ray/tests/test_node_label_scheduling_strategy.py

```diff
@@ -28,24 +32,124 @@ def double(self, x):
 )
 def test_node_label_scheduling_basic(call_ray_start):
     ray.init(address=call_ray_start)
-    MyActor.options(
+    actor = MyActor.options(
         scheduling_strategy=NodeLabelSchedulingStrategy(
             {"gpu_type": In("A100", "T100"), "region": Exists()}
         )
-    )
+    ).remote()
+    assert ray.get(actor.value.remote(), timeout=3) == 0
+
+    actor = MyActor.options(
+        scheduling_strategy=NodeLabelSchedulingStrategy({"gpu_type": NotIn("A100")})
+    ).remote()
+    with pytest.raises(TimeoutError):
+        assert ray.get(actor.value.remote(), timeout=3) == 0
 
-    MyActor.options(
+    actor = MyActor.options(
         scheduling_strategy=NodeLabelSchedulingStrategy(
-            {"gpu_type": NotIn("A100", "T100"), "other_key": DoesNotExist()}
+            hard={"gpu_type": DoesNotExist()},
+            soft={"gpu_type": In("A100")},
         )
-    )
+    ).remote()
+    with pytest.raises(TimeoutError):
+        assert ray.get(actor.value.remote(), timeout=3) == 0
 
-    MyActor.options(
+    actor = MyActor.options(
         scheduling_strategy=NodeLabelSchedulingStrategy(
-            hard={"gpu_type": Exists()},
-            soft={"gpu_type": In("A100")},
+            hard={"gpu_type": In("T100")},
+        )
+    ).remote()
+    with pytest.raises(TimeoutError):
+        assert ray.get(actor.value.remote(), timeout=3) == 0
+
+    actor = MyActor.options(
+        scheduling_strategy=NodeLabelSchedulingStrategy(
+            hard={},
+            soft={"gpu_type": In("T100ssss")},
+        )
+    ).remote()
+    assert ray.get(actor.value.remote(), timeout=3) == 0
+
+
+def test_node_label_scheduling_in_cluster(ray_start_cluster):
```

Comment:
We should add more tests like: doesn't match soft but can still be scheduled.

Reply:
Yes, i will add a TODO.